first_diff.py

```diff
@@ -1,255 +1,55 @@
 #!/usr/bin/env python3
 
-import os.path
+from __future__ import annotations
```

Comment:
I'm not sure why this is here

It changes the behavior of annotations but I don't think you care here
(And if I missed something and you do I'd suggest just using string type hints instead)
https://stackoverflow.com/questions/61544854/from-future-import-annotations